Src/FluentAssertions/Common/TypeMemberReflector.cs

```diff
@@ -31,27 +31,39 @@ private static PropertyInfo[] LoadNonPrivateProperties(Type typeToReflect, Membe
     {
         IEnumerable<PropertyInfo> query =
             from propertyInfo in GetPropertiesFromHierarchy(typeToReflect, visibility)
-            where HasNonPrivateGetter(propertyInfo)
+            where HasNonPrivateGetter(propertyInfo, visibility)
             where !propertyInfo.IsIndexer()
             select propertyInfo;
 
         return query.ToArray();
     }
 
-    private static List<PropertyInfo> GetPropertiesFromHierarchy(Type typeToReflect, MemberVisibility memberVisibility)
+    private static IEnumerable<PropertyInfo> GetPropertiesFromHierarchy(Type typeToReflect, MemberVisibility memberVisibility)
     {
-        bool includeInternals = memberVisibility.HasFlag(MemberVisibility.Internal);
+        bool includeInternal = memberVisibility.HasFlag(MemberVisibility.Internal);
 
         return GetMembersFromHierarchy(typeToReflect, type =>
         {
             return type
                 .GetProperties(AllInstanceMembersFlag | BindingFlags.DeclaredOnly)
-                .Where(property => property.GetMethod?.IsPrivate == false)
-                .Where(property => includeInternals || property.GetMethod is { IsAssembly: false, IsFamilyOrAssembly: false })
+                .Where(property => includeInternal || !IsInternal(property))
+                .OrderBy(property => IsExplicitImplementation(property))
                 .ToArray();
         });
     }
 
+    private static bool IsInternal(PropertyInfo property)
+    {
+        return property.GetMethod is { IsAssembly: true } or { IsFamilyOrAssembly: true };
+    }
+
+    private static bool IsExplicitImplementation(PropertyInfo property)
+    {
+        return property.GetMethod?.IsPrivate == true &&
+            property.SetMethod?.IsPrivate == true &&
+            property.Name.Contains('.', StringComparison.Ordinal);
+    }
```

Comment:
Needs to be `property.SetMethod?.IsPrivate != false` to account for explicitly implemented properties without a setter.

If we're okay with letting this method know that we're filtered out properties without getters earlier (the commend above), we can remove the `?` from `GetMethod`

Reply:
Added an extra test to cover this scenario.